src/main/scala/nl.knaw.dans.easy.validatebag/rules/metadata/package.scala

```diff
@@ -389,6 +390,24 @@ package object metadata extends DebugEnhancedLogging {
     }
   }
 
+  def filesXmlFilesHaveOnlyAllowedAccessRights(t: TargetBag): Try[Unit] = {
+    trace(())
+      for {
+        xml <- t.tryFilesXml
+        _ <- getAccessRightElements(xml).map(validateAccessRights).collectResults.recoverWith {
```

Comment:
1. I didn't mean to extract just `xml \ "file" \ "accessRights"` into a method, but rather the whole thing, particularly the `map`, `collectResults` and `recoverWith`.
2. don't query for `xml \ "file" \ "accessRights"`, but rather `xml \ "file"` and pass each `<file/>` to `validateAccessRights`. Then in that method, do something like `fileNode \@ "filepath"` and `fileNode \ "accessRights"`. Don't think too complicated!